apps/music/test/marionette/playlist_test.js

```diff
@@ -358,5 +358,54 @@ marionette('Music player playlist', function() {
       assert.equal(PlaylistHelper.songTitle(songs[1]), 'Yield to thread');
     });
 
+    test('Shuffle all sort order', function() {
+      music.launch();
+      music.waitForFirstTile();
+
+      music.switchToPlaylistsView();
+
+      var notrandom = 0;
+      var lastTitle = '';
+      for (var i = 0; i < 10; i++) {
+
+        music.selectPlaylist('Shuffle all');
+
+        var title = music.title.text();
+        if (title === lastTitle) {
```

Comment:
The first time through this loop, lastTitle will be the empty string and title will presumably not be the empty string. You need to initialize lastTitle to something, or ignore this first iteration, I think.